src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java

```diff
@@ -335,24 +342,42 @@ private void appendMacroSpecificCompletions(String input, List<Completion> resul
 				String command = trimmedline.substring(8).trim().replace("{", "");
 				String lcasecommand = command.toLowerCase();
 				if (lcasecommand.contains(lcaseinput)) {
-					String description = findDescription(textArray, linecount, "User defined function " + command + "\n as specified in line " + (linecount + 1));
+					String description = findDescription(textArray, linecount, "User defined function <b>" + command + "</b> as specified in line " + (linecount + 1));
 
 
 					completions.add(new BasicCompletion(this, command, null, description));
 				}
 			}
-			if (lcaseline.contains("=")) {
+			if ((lcaseline.contains("=")) && (charcount < codeLength)) {
```

Comment:
Hi @haesleinhuepf, thanks for taking the time to review! 

Regarding this question: the text that's been scanned comprises the code in the editor (which spans `codeLength` characters), plus the block of additional functions concatenated at the end. Then, if `charcount` --the running counter of processed characters-- surpasses `codeLength`, the code has started to scan the additional functions, where variable names are irrelevant to autocompletition.  Perhaps a short comment can help clarify this:

```suggestion
			if ((lcaseline.contains("=")) && (charcount < codeLength)) { // possible variable assignment AND within user code block (not in additional functions)
```

On a more general note, I tried to keep the code comments to a bare minimum, as I saw they were sparsely present, but maybe I can be more verbose each of the modifications, so that anyone (including future me) can take a glimpse and have an fast overall idea of the logic that's (supossedly) implemented. Just let me know, and I can walk back most of the commits and document as needed.